examples/ffi/profile_intern.cpp

```diff
@@ -0,0 +1,146 @@
+    }                                                                                              \
+  }
+
+CHECK_RESULT(ddog_VoidResult, DDOG_VOID_RESULT_OK)
+
+#define EXTRACT_RESULT(typ, uppercase)                                                             \
+  ddog_prof_##typ##Id extract_result(ddog_prof_##typ##Id_Result result) {                          \
+    if (result.tag != DDOG_PROF_##uppercase##_ID_RESULT_OK_GENERATIONAL_ID_##uppercase##_ID) {     \
+      print_error(result.err);                                                                     \
+      ddog_Error_drop(&result.err);                                                                \
+      exit(EXIT_FAILURE);                                                                          \
+    } else {                                                                                       \
+      return result.ok;                                                                            \
+    }                                                                                              \
+  }
+
+EXTRACT_RESULT(Function, FUNCTION)
+EXTRACT_RESULT(Label, LABEL)
+EXTRACT_RESULT(LabelSet, LABEL_SET)
+EXTRACT_RESULT(Location, LOCATION)
+EXTRACT_RESULT(Mapping, MAPPING)
+EXTRACT_RESULT(StackTrace, STACK_TRACE)
+EXTRACT_RESULT(String, STRING)
+
+void wait_for_user(std::string s) {
+  std::cout << s << std::endl;
+  getchar();
+}
+
+int main(void) {
+  const ddog_prof_ValueType wall_time = {
+      .type_ = DDOG_CHARSLICE_C("wall-time"),
+      .unit = DDOG_CHARSLICE_C("nanoseconds"),
+  };
+  const ddog_prof_Slice_ValueType sample_types = {&wall_time, 1};
+  const ddog_prof_Period period = {wall_time, 60};
+
+  ddog_prof_Profile_NewResult new_result = ddog_prof_Profile_new(sample_types, &period, NULL);
+  if (new_result.tag != DDOG_PROF_PROFILE_NEW_RESULT_OK) {
+    ddog_CharSlice message = ddog_Error_message(&new_result.err);
+    fprintf(stderr, "%.*s", (int)message.len, message.ptr);
+    ddog_Error_drop(&new_result.err);
+    exit(EXIT_FAILURE);
+  }
+
+  ddog_prof_Profile *profile = &new_result.ok;
+  auto empty_string_id =
+      extract_result(ddog_prof_Profile_intern_string(profile, DDOG_CHARSLICE_C("")));
+
+  auto root_function_name =
+      extract_result(ddog_prof_Profile_intern_string(profile, DDOG_CHARSLICE_C("{main}")));
+  auto root_file_name = extract_result(
+      ddog_prof_Profile_intern_string(profile, DDOG_CHARSLICE_C("/srv/example/index.php")));
+  auto root_mapping = extract_result(
+      ddog_prof_Profile_intern_mapping(profile, 0, 0, 0, root_file_name, empty_string_id));
+  auto root_function = extract_result(ddog_prof_Profile_intern_function(
+      profile, root_function_name, empty_string_id, root_file_name, 0));
+  auto root_location =
+      extract_result(ddog_prof_Profile_intern_location(profile, root_mapping, root_function, 0, 0));
+  ddog_prof_Slice_GenerationalIdLocationId locations = {.ptr = &root_location, .len = 1};
```

Comment:
Minor: Should probably be renamed to `ddog_prof_Slice_LocationId`

Reply:
done